source/extensions/filters/http/json_to_metadata/filter.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.pb.h"
+#include "envoy/json/json_object.h"
+#include "envoy/server/filter_config.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JsonToMetadata {
+
+// clang-format off
+#define ALL_JSON_TO_METADATA_FILTER_STATS(COUNTER)                \
+  COUNTER(rq_success)                                             \
+  COUNTER(rq_mismatched_content_type)                             \
+  COUNTER(rq_no_body)                                             \
+  COUNTER(rq_too_large_body)                                      \
+  COUNTER(rq_invalid_json_body)
+// clang-format on
+
+struct JsonToMetadataStats {
+  ALL_JSON_TO_METADATA_FILTER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+using ProtoRule = envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::Rule;
+using KeyValuePair =
+    envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::KeyValuePair;
+using ValueType = envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::ValueType;
+
+class Rule {
```

Comment:
consider just making this a struct with the keys_ vector const. Then we could get rid of getters and the entire object would be "const"

Reply:
yep it's 
```
using Rules = std::vector<Rule>;

Rules request_rules_;
```
in the member so i'll introduce a helper function to make `const Rules request_rules_`